PIL/GifImagePlugin.py

```diff
@@ -453,15 +453,24 @@ def _get_local_header(fp, im, offset, flags):
         duration = int(im.encoderinfo["duration"] / 10)
     else:
         duration = 0
+
+    if "disposal" in im.encoderinfo:
+        disposal = int(im.encoderinfo['disposal'])
+        if not 0<= disposal <=3:
+            disposal = 2
+    else:
+        disposal = 2
+
     if transparent_color_exists or duration != 0:
```

Comment:
I don't think that this is the correct case anymore -- unless disposal is only valid with a transparent color or duration.